src/app/static/src/app/store/root/getters.ts

```diff
@@ -1,12 +1,63 @@
 import {RootState} from "../../root";
 import {Getter, GetterTree} from "vuex";
+import {Error} from "../../generated"
 
 interface RootGetters {
     isGuest: Getter<RootState, RootState>
 }
 
 export const getters: RootGetters & GetterTree<RootState, RootState> = {
-    isGuest: (state: RootState, getters: any) => {
+    isGuest: (state: RootState) => {
         return state.currentUser == "guest";
+    },
+    errors: (state: RootState) => {
+        const {
+            adr,
+            adrUpload,
+            baseline,
+            downloadResults,
+            load,
+            metadata,
+            modelCalibrate,
+            modelOptions,
+            modelOutput,
+            plottingSelections,
+            projects,
+            surveyAndProgram
+        } = state;
+
+        return ([] as Error[]).concat.apply([] as Error[], [extractErrors(adr),
+            extractErrors(adrUpload),
+            extractErrors(baseline),
+            extractErrors(downloadResults),
+            extractErrors(load),
+            extractErrors(metadata),
+            extractErrors(modelCalibrate),
+            extractErrors(modelOptions),
+            extractErrors(modelOutput),
+            extractErrors(plottingSelections),
+            extractErrors(projects),
+            extractErrors(surveyAndProgram),
+            state.modelRun.errors,
+            state.errors.errors]);
+    }
+}
+
+const extractErrors = (state: any) => {
```

Comment:
I'm afraid I would be keen to see a test specifically for this slightly gnarly function - not least as a means of documenting its intended behaviour. I know you're testing it indirectly already, but given it's standalone it should be quite easy to add a further test to ensure that things with and without `[Ee]rror` suffix, and without and without an `error` field get in/excluded appropriately.

Reply:
https://github.com/mrc-ide/hint/pull/562/commits/2d0cc813d499ae15a78a48842c9abf29b1b4b696